bootloader/grub.go

```diff
@@ -80,24 +79,34 @@ func (g *grub) dir() string {
 	return filepath.Join(g.rootdir, g.basedir)
 }
 
-func (g *grub) installRecoveryBootConfig(gadgetDir string) (bool, error) {
+func (g *grub) installManagedRecoveryBootConfig(gadgetDir string) (bool, error) {
 	gadgetGrubCfg := filepath.Join(gadgetDir, g.Name()+".conf")
 	if !osutil.FileExists(gadgetGrubCfg) {
 		// gadget does not use grub bootloader
 		return false, nil
 	}
 	assetName := g.Name() + "-recovery.cfg"
-	bootConfig := assets.Internal(assetName)
-	if bootConfig == nil {
-		return true, fmt.Errorf("internal error: no boot asset for %q", assetName)
+	systemFile := filepath.Join(g.rootdir, "/EFI/ubuntu/grub.cfg")
+	return genericSetBootConfigFromAsset(systemFile, assetName)
+}
+
+func (g *grub) installManagedBootConfig(gadgetDir string) (bool, error) {
+	gadgetGrubCfg := filepath.Join(gadgetDir, g.Name()+".conf")
+	if !osutil.FileExists(gadgetGrubCfg) {
+		// gadget does not use grub bootloader
+		return false, nil
 	}
+	assetName := g.Name() + ".cfg"
 	systemFile := filepath.Join(g.rootdir, "/EFI/ubuntu/grub.cfg")
-	return genericSetBootConfig(systemFile, bootConfig)
+	return genericSetBootConfigFromAsset(systemFile, assetName)
 }
 
 func (g *grub) InstallBootConfig(gadgetDir string, opts *Options) (bool, error) {
 	if opts != nil && opts.Recovery {
-		return g.installRecoveryBootConfig(gadgetDir)
+		return g.installManagedRecoveryBootConfig(gadgetDir)
+	}
+	if opts != nil && opts.NoSlashBoot {
```

Comment:
this should use maybe use ExtractedRunKernelImage, no?  either way it probably needs a TODO:UC20 about finding better names for the bootloader flags,  we have a similar problem already in bootstate20.go in bootState20Kernel.loadBootenv

Reply:
Perhaps it should be both. NoSlashBoot because we are looking at the native hierarchy under the root of the boot partition, and ExtractedRunKernelImage because the script must be aware of a single kernel.efi file?

Reply:
both is fine until we rename them